hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1801,6 +1802,22 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		if err := r.Client.Status().Update(ctx, hcluster); err != nil {
 			return ctrl.Result{}, fmt.Errorf("failed to update status: %w", err)
 		}
+	case hyperv1.KubevirtPlatform:
+		if hcluster.Spec.Platform.Kubevirt.BaseDomainPassthrough != nil && *hcluster.Spec.Platform.Kubevirt.BaseDomainPassthrough {
+			machines, err := r.listMachinesForCluster(hcluster)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed retrieving machines to compose default ingress passthrough service endpoints: %w", err)
+			}
+			kvInfraClient, err := r.KubevirtInfraClients.DiscoverKubevirtClusterClient(ctx, r.Client, hcluster.Spec.InfraID, hcluster.Spec.Platform.Kubevirt.Credentials, hcluster.Namespace, hcluster.Namespace)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed discovering kubevirt cluster client when creating ingress endpoints: %w", err)
+			}
+			if err := kubevirt.ReconcileDefaultIngressEndpoints(ctx, kvInfraClient.GetInfraClient(), hcluster, machines); err != nil {
+				log.Error(err, "failed reconciling default ingress passthrough service endpoints")
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+			}
```

Comment:
Is kind of chicken egg, have to check if doing it so early prevent the config-operator (the one that creates the service) to start up.

I am thinking that an alternative is to add a controller to config operator similar to the one we have for have for node but for machines, it will just fail on conflict so will be able to update the endpoints, I think we don't really care for nodepools we care for Machine, Service and HostedCluster.

HostedCluster controller is already quite complex.